versioned/mongodb/src/test/java/com/dremio/nessie/versioned/store/mongodb/TestMongoDbStore.java

```diff
@@ -0,0 +1,274 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.store.mongodb;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.dremio.nessie.versioned.impl.Fragment;
+import com.dremio.nessie.versioned.impl.InternalCommitMetadata;
+import com.dremio.nessie.versioned.impl.InternalRef;
+import com.dremio.nessie.versioned.impl.InternalValue;
+import com.dremio.nessie.versioned.impl.L1;
+import com.dremio.nessie.versioned.impl.L2;
+import com.dremio.nessie.versioned.impl.L3;
+import com.dremio.nessie.versioned.store.HasId;
+import com.dremio.nessie.versioned.store.SaveOp;
+import com.dremio.nessie.versioned.store.SimpleSchema;
+import com.dremio.nessie.versioned.store.ValueType;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.mongodb.ConnectionString;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+
+import de.flapdoodle.embed.mongo.MongodExecutable;
+import de.flapdoodle.embed.mongo.MongodStarter;
+import de.flapdoodle.embed.mongo.config.MongodConfig;
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.distribution.Version;
+import de.flapdoodle.embed.process.runtime.Network;
+
+public class TestMongoDbStore {
+  private static MongodExecutable mongoExec;
+  private static ConnectionString connectionString;
+  private static final String testDatabaseName = "mydb";
+  private MongoDbStore mongoDbStore;
+
+  /**
+   * Set up the embedded flapdoodle MongoDB server for unit tests.
+   * @throws IOException if there's an issue grabbing the port or determining IP version.
+   */
+  @BeforeAll
+  public static void setupServer() throws IOException {
+    final int port = Network.getFreeServerPort();
+    final MongodConfig config = MongodConfig.builder()
+        .version(Version.Main.PRODUCTION)
+        .net(new Net(port, Network.localhostIsIPv6()))
+        .build();
+
+    mongoExec = MongodStarter.getDefaultInstance().prepare(config);
+    mongoExec.start();
+    connectionString = new ConnectionString("mongodb://localhost:" + port);
+  }
+
+  /**
+   * Shut down the embedded flapdoodle MongoDB server.
+   */
+  @AfterAll
+  public static void teardownServer() {
+    if (null != mongoExec) {
+      mongoExec.stop();
+    }
+  }
+
+  /**
+   * Set up the objects necessary for the tests.
+   */
+  @BeforeEach
+  public void setUp() {
+    mongoDbStore = new MongoDbStore(connectionString, testDatabaseName);
+    mongoDbStore.start();
```

Comment:
can we simply clean between tests instead of start stop (i'm assuming the latter will make the tests slower)